### PR TITLE
Add action to get latest release

### PR DESCRIPTION
**Pull request by lfkellogg**
Addresses https://github.com/fastlane/fastlane-plugin-firebase_app_distribution/issues/25.

Example usage to increment build number based on last release:
```
lane :bump_version do
  latest_release = firebase_app_distribution_get_latest_release(
    app: "1:1234567890:ios:0a1b2c3d4e5f67890"
  )
  increment_build_number({ build_number: latest_release[:buildVersion].to_i + 1 })
end
```
          
Fastlane action docs:
```
$ bundle exec fastlane action firebase_app_distribution_get_latest_release
[✔] 🚀
+-------------------------------------------+---------+------------------------------------------------+
|                                             Used plugins                                             |
+-------------------------------------------+---------+------------------------------------------------+
| Plugin                                    | Version | Action                                         |
+-------------------------------------------+---------+------------------------------------------------+
| fastlane-plugin-firebase_app_distribution | 0.3.1   | firebase_app_distribution_login,               |
|                                           |         | firebase_app_distribution_add_testers,         |
|                                           |         | firebase_app_distribution_get_udids,           |
|                                           |         | firebase_app_distribution,                     |
|                                           |         | firebase_app_distribution_get_latest_release,  |
|                                           |         | firebase_app_distribution_remove_testers       |
+-------------------------------------------+---------+------------------------------------------------+

Loading documentation for firebase_app_distribution_get_latest_release:

+-------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                     firebase_app_distribution_get_latest_release                                                      |
+-------------------------------------------------------------------------------------------------------------------------------------------------------+
| Fetches the latest release in Firebase App Distribution                                                                                               |
|                                                                                                                                                       |
| Fetches information about the most recently created release in App Distribution, including the version and release notes. Returns nil if no releases  |
| are found.                                                                                                                                            |
|                                                                                                                                                       |
| Created by lkellogg@google.com                                                                                                                        |
+-------------------------------------------------------------------------------------------------------------------------------------------------------+

+--------------------------+------------------------------------+-----------------------+---------+
|                      firebase_app_distribution_get_latest_release Options                       |
+--------------------------+------------------------------------+-----------------------+---------+
| Key                      | Description                        | Env Var(s)            | Default |
+--------------------------+------------------------------------+-----------------------+---------+
| app                      | Your app's Firebase App ID. You    | FIREBASEAPPDISTRO_APP |         |
|                          | can find the App ID in the         |                       |         |
|                          | Firebase console, on the General   |                       |         |
|                          | Settings page                      |                       |         |
| firebase_cli_token       | Auth token generated using         |                       |         |
|                          | 'fastlane run                      |                       |         |
|                          | firebase_app_distribution_login',  |                       |         |
|                          | or the Firebase CLI's login:ci     |                       |         |
|                          | command                            |                       |         |
| service_credentials_file | Path to Google service account     |                       |         |
|                          | json                               |                       |         |
| debug                    | Print verbose debug output         |                       | false   |
+--------------------------+------------------------------------+-----------------------+---------+
* = default value is dependent on the user's system

+------------------------------------+------------------------------------------------------------------+
|                     firebase_app_distribution_get_latest_release Output Variables                     |
+------------------------------------+------------------------------------------------------------------+
| Key                                | Description                                                      |
+------------------------------------+------------------------------------------------------------------+
| FIREBASE_APP_DISTRO_LATEST_RELEASE | A hash representing the lastest release created in Firebase App  |
|                                    | Distribution                                                     |
+------------------------------------+------------------------------------------------------------------+
Access the output values using `lane_context[SharedValues::VARIABLE_NAME]`

+---------------------------------------------------------------------------------------------------------------+
|                           firebase_app_distribution_get_latest_release Return Value                           |
+---------------------------------------------------------------------------------------------------------------+
| Hash representation of the lastest release created in Firebase App Distribution (see                          |
| https://firebase.google.com/docs/reference/app-distribution/rest/v1/projects.apps.releases#resource:-release) |
+---------------------------------------------------------------------------------------------------------------+
```

### CODE REVIEW

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      extend Auth::FirebaseAppDistributionAuthClient
+      extend Helper::FirebaseAppDistributionHelper
+
+      def self.run(params)
+        auth_token = fetch_auth_token(params[:service_credentials_file], params[:firebase_cli_token])
+        fad_api_client = Client::FirebaseAppDistributionApiClient.new(auth_token, params[:debug])
+
+        UI.message("⏳ Fetching latest release for app #{params[:app]}...")
+
+        releases = fad_api_client.list_releases(app_name_from_app_id(params[:app]), 1)[:releases] || []
```
Comment (tagboola) [inline lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:18]: Consider handling the app name parsing in the API client

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        if releases.length == 0
+         	UI.user_error!("No releases for app #{params[:app]} found in App Distribution.")
```
Comment (tagboola) [inline lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:20]: Hmm I wonder if we shouldn't error here. A possible valid use case is to try and get the latest release only to find out there isn't one, and then they could just move forward and upload the binary.

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
+require 'fastlane/action'
+require_relative '../client/firebase_app_distribution_api_client'
+require_relative '../helper/firebase_app_distribution_auth_client'
+require_relative '../helper/firebase_app_distribution_helper'
+
+module Fastlane
+  module Actions
+    class FirebaseAppDistributionGetLatestReleaseAction < Action
```
Comment (tagboola) [inline lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:8]: Consider adding a specs for the action

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb (hunk 2 above)
Comment (lfkellogg) [reply lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:20]: So we could return `nil` in that case? That sounds reasonable to me.

Comment (tagboola) [reply lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:20]: SGTM!

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb (hunk 3 above)
Comment (lfkellogg) [reply lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:8]: Done! Also added some for the new API client method.

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb (hunk 1 above)
Comment (tagboola) [reply lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_get_latest_release.rb:18]: 👍

**Review by tagboola — APPROVED** [review]
🚢

**Pull request merged**